instrumentation-core/reactor-3.1/src/main/java/io/opentelemetry/instrumentation/reactor/TracingSubscriber.java

```diff
@@ -18,132 +18,65 @@
 
 import io.opentelemetry.context.ContextUtils;
 import io.opentelemetry.context.Scope;
+import org.reactivestreams.Subscriber;
 import org.reactivestreams.Subscription;
 import reactor.core.CoreSubscriber;
-import reactor.core.Fuseable;
-import reactor.core.Scannable;
 import reactor.util.context.Context;
 
-public class TracingSubscriber<T>
-    implements Subscription, CoreSubscriber<T>, Fuseable.QueueSubscription<T>, Scannable {
-
-  private final io.grpc.Context upstreamContext;
-  private final CoreSubscriber<T> delegate;
+/**
+ * Based on Spring Sleuth's Reactor instrumentation. A trace representation of the {@link
+ * Subscriber}
+ *
+ * @author Jose Montoya
```

Comment:
Since we don't have `@author` anywhere else I would recommend not including it here too even for copied files.

Ideally we fix spotless to allow us to have two copyright headers on a file. In the meantime, I took a NOTICE approach to referencing copied code which I think is the bare minimum legally (`@author` doesn't help at all legally)

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/instrumentation/logback/logback-1.0.0/library/NOTICE.txt